### PR TITLE
[Chips] Delete Themers

### LINKED ISSUE #9027: [Chips] Delete Chips Themer classes
These classes are deprecated and have no internal usage. 

---
#### Internal data
- Associated internal bug: [b/145233378](http://b/145233378)

### PR DESCRIPTION
**Pull request by bryanoltman**
This PR moves logic from the Chips Themer classes into the Chip view theming extension (MDCChipView+MaterialTheming.m) and deletes the unused Themer classes. 

Fixes #9027

### CODE REVIEW

File: components/Chips/src/ColorThemer/MDCChipViewColorThemer.h [hunk 1]
```diff
@@ -1,72 +0,0 @@
...
-
- @warning This API will eventually be deprecated. See the individual method documentation for
- details on replacement APIs.
- Learn more at docs/theming.md#migration-guide-themers-to-theming-extensions
- */
-@interface MDCChipViewColorThemer : NSObject
-
-@end
-
-@interface MDCChipViewColorThemer (ToBeDeprecated)
```
Comment (romoore) [inline components/Chips/src/ColorThemer/MDCChipViewColorThemer.h]: Please do not delete this class (or any other class) until it's been explicitly marked deprecated for one release version.

**Review by romoore — CHANGES_REQUESTED** [review]
Generally in favor, but some of these APIs were not deprecated and [need to be marked deprecated for at least one release version](https://github.com/material-components/material-components-ios/blob/cb6b8e9599e9018c545e0011a3ded82597988106/contributing/deprecation_policy.md#our-deprecation-process-googlers) before we delete them.

**Comment by bryanoltman** [comment]
@romoore All of the classes touched by this PR are marked deprecated in today's release.

File: components/Chips/src/ColorThemer/MDCChipViewColorThemer.h (hunk 1 above)
Comment (bryanoltman) [reply components/Chips/src/ColorThemer/MDCChipViewColorThemer.h]: Good catch. These are marked deprecated in 94.5.0, whose release is currently in progress.

File: components/schemes/Shape/examples/supplemental/MDCChipViewShapeThemerDefaultMapping.m [hunk 2]
```diff
@@ -1,29 +0,0 @@
...
-// limitations under the License.
-
-#import "MDCChipViewShapeThemerDefaultMapping.h"
-
-@implementation MDCChipViewShapeThemerDefaultMapping
-
-+ (void)applyShapeScheme:(nonnull id<MDCShapeScheming>)shapeScheme
-              toChipView:(nonnull MDCChipView *)chipView {
-  MDCRectangleShapeGenerator *rectangleShape = [[MDCRectangleShapeGenerator alloc] init];
-  rectangleShape.topLeftCorner = shapeScheme.smallComponentShape.topLeftCorner;
```
Comment (romoore) [inline components/schemes/Shape/examples/supplemental/MDCChipViewShapeThemerDefaultMapping.m:22]: Please move this behavior into the example VC so we can preserve showing how to apply Shapes to Chips.

**Review by romoore — APPROVED** [review]
LGTM assuming the chip shape theming *behavior* is preserved.

**Pull request merged**